addons/web/static/src/model/relational_model/dynamic_list.js

```diff
@@ -349,8 +349,11 @@ export class DynamicList extends DataPoint {
         }
 
         // Perform the resequence in the list of records/groups
-        const [dp] = originalList.splice(fromIndex, 1);
-        originalList.splice(toIndex, 0, dp);
+        const dp = originalList[fromIndex];
+        if (fromIndex !== toIndex) {
+            originalList.splice(fromIndex, 1);
+            originalList.splice(toIndex, 0, dp);
```

Comment:
```js
[originalList[fromIndex], originalList[toIndex]] = [originalList[toIndex], originalList[fromIndex]];
```
Maybe? I think it communicates intent better and is likely faster because most items don't need to move.